Ezzywealth/cab_bookings_api
# Add signin authentication

## Description
This pull request adds sign-in authentication using JWT (JSON Web Tokens) to the application. It enhances the security and user authentication process by generating and validating tokens for authenticated users.

## Changes Made
- Added the necessary gems and configurations for JWT authentication.
- Implemented JWT token generation upon successful sign-in.
- Implemented JWT token validation and user retrieval from the token for authenticated routes.
- Updated the `SessionsController` to handle JWT authentication and respond with the generated token upon successful sign-in.
- Modified the `routes.rb` file to include the necessary routes for JWT authentication.

## How to Test
1. Start the application and ensure the server is running.
2. Use an API testing tool (e.g., Postman) to send a `POST` request to the `/api/v1/users/sign_in` endpoint with valid user credentials.
3. Verify that the response includes a JSON object containing a `token` field.
4. Include the generated token in the headers of subsequent requests to authenticated routes.
5. Verify that accessing authenticated routes without a valid token results in a `401 Unauthorized` response.
6. Verify that accessing authenticated routes with a valid token succeeds and returns the expected data.

## Checklist
- [ ] I have tested the changes and verified their functionality.
- [ ] I have reviewed the code and ensured its quality.
- [ ] I have added appropriate test cases.
